### PR TITLE
Add rings to the Warden Aura

### PR DESCRIPTION
**Pull request by MrSmig**
Also fix the Warden's "Deus Ex Machina" perk giving the "Rejection Pulse" shockwave effect to newly bought towers.

### CODE REVIEW

File: gamemodes/horde/gamemode/cl_stencil_rings.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	rings.RenderSphere( zones )
+
+	rings.CamRendering( LocalPlayer():EyeAngles():Forward(), col )
+	
+	render.SetStencilEnable( false )
+end
+
+hook.Add("PostDrawOpaqueRenderables","RenderRings",function()
+	List = {}
+	col = Color(255,255,255,255)
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/cl_stencil_rings.lua:67]: you shouldnt recreate the color here each time

File: gamemodes/horde/gamemode/cl_stencil_rings.lua [hunk 2]
```diff
@@ -0,0 +1,72 @@
+local colmask = Color(0,0,0,0)
+local render_DrawSphere = render.DrawSphere
+
+rings = rings or {}
+
+local List = {}
+local col = Color(255,255,255,255)
+
+function rings.SetColor(color)
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/cl_stencil_rings.lua:9]: considering setcolor doesnt really work for multiple colors right now, leave a comment or so, so we at least know about it in the future

File: gamemodes/horde/gamemode/cl_stencil_rings.lua [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	local cam_pos = EyePos( LocalPlayer() )
+	render.SetStencilCompareFunction( STENCIL_EQUAL )
+	render.SetStencilReferenceValue( 1 )
+	render.DrawQuadEasy(cam_pos + cam_normal*10, -cam_normal, 100, 200, Color, 0)
+end
+
+function rings.RenderRings( zones, col )
+
+	if next(zones) == nil then return end
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/cl_stencil_rings.lua:55]: you probably dont have to check for this, as you already check the list

File: gamemodes/horde/gamemode/status/buff/sv_warden_aura.lua [hunk 4]
```diff
@@ -1,8 +1,11 @@
 local entmeta = FindMetaTable("Entity")
 local plymeta = FindMetaTable("Player")
 
+util.AddNetworkString("AddNewAuraRing")
+
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/status/buff/sv_warden_aura.lua:5]: nitpick, you dont need a newline here

File: gamemodes/horde/entities/entities/horde_warden_aura/cl_init.lua [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+function ENT:OnRemove()
+	auras[self] = nil
+end
+
+local col = Color(0,255,100,100)
+hook.Add( "PreDrawRings", "DrawWardenRings", function()
+	if next( auras ) == nil then return end
+	rings.SetColor(col)
+	for ent,bool in pairs( auras ) do
+		if !IsValid(ent) then continue end
```
Comment (wrefgtzweve) [inline gamemodes/horde/entities/entities/horde_warden_aura/cl_init.lua:18]: use `not` instead of `!`

**Review by wrefgtzweve — APPROVED** [review]

File: gamemodes/horde/gamemode/status/buff/sv_warden_aura.lua (hunk 4 above)
Comment (MrSmig) [reply gamemodes/horde/gamemode/status/buff/sv_warden_aura.lua:5]: I forgot above those, they aren't even needed at all anymore

**Pull request merged**